synth/src/datasource/postgres_datasource.rs

```diff
@@ -0,0 +1,330 @@
+            FROM information_schema.table_constraints AS tc 
+            JOIN information_schema.key_column_usage AS kcu 
+            ON tc.constraint_name = kcu.constraint_name
+            JOIN information_schema.constraint_column_usage AS ccu 
+            ON ccu.constraint_name = tc.constraint_name
+            WHERE constraint_type = 'FOREIGN KEY'";
+
+        sqlx::query(query)
+            .fetch_all(&self.pool)
+            .await?
+            .into_iter()
+            .map(ForeignKey::try_from)
+            .collect::<Result<Vec<ForeignKey>>>()
+    }
+
+    /// Must use the singled threaded pool when setting this in conjunction with random, called by
+    /// [get_deterministic_samples]. Otherwise, expect endless facepalms (-_Q)
+    async fn set_seed(&self) -> Result<()> {
+        sqlx::query("SELECT setseed(0.5)").execute(&self.single_thread_pool).await?;
+        Ok(())
+    }
+
+    /// Must use the singled threaded pool when setting this in conjunction with setseed, called by
+    /// [set_seed]. Otherwise, expect big regrets :(
+    async fn get_deterministic_samples(&self, table_name: &str) -> Result<Vec<Value>> {
+        let query: &str = &format!("SELECT * FROM {} ORDER BY random() LIMIT 10", table_name);
+
+        let values = sqlx::query(query)
+            .fetch_all(&self.single_thread_pool)
+            .await?
+            .into_iter()
+            .map(ValueWrapper::try_from)
+            .map(|v| {
+                match v {
+                    Ok(wrapper) => Ok(wrapper.0),
+                    Err(e) => bail!("Failed to convert to value wrapper from query results: {:?}", e)
+                }
+            })
+            .collect::<Result<Vec<Value>>>()?;
+
+        Ok(values)
+    }
+
+    fn decode_to_content(&self, data_type: &str, char_max_len: Option<i32>) -> Result<Content> {
+        let content = match data_type.to_lowercase().as_str() {
+            "bool" => Content::Bool(BoolContent::default()),
+            "oid" => {
+                bail!("OID data type not supported")
+            }
+            "char" | "varchar" | "text" | "bpchar" | "name" | "unknown" => {
+                let pattern = "[a-zA-Z0-9]{0, {}}".replace(
+                    "{}",
+                    &format!("{}", char_max_len.unwrap_or(1)),
+                );
+                Content::String(StringContent::Pattern(
+                    RegexContent::pattern(pattern).context("pattern will always compile")?,
+                ))
+            }
+            "int2" => Content::Number(NumberContent::I64(I64::Range(RangeStep {
+                low: 0,
+                high: 1,
+                step: 1,
+            }))),
+            "int4" => Content::Number(NumberContent::I64(I64::Range(RangeStep {
+                low: 0,
+                high: 1,
+                step: 1,
+            }))),
+            "int8" => Content::Number(NumberContent::I64(I64::Range(RangeStep {
+                low: 1,
```

Comment:
`low` needs to be explicitly lower than high for all ranges.

You can change it to `low: 0` and it should work :)